components/channel_layout/channel_controller.test.tsx

```diff
@@ -2,99 +2,54 @@
-        Object.defineProperty(window.navigator, 'platform', {
-            value: 'Win32',
-            writable: true,
-        });
-        shallow(<ChannelController {...props}/>);
-
-        expect(document.body.classList.contains('os--windows')).toBe(true);
-    });
-
-    test('Should have os--windows class on body for windows 64', () => {
-        Object.defineProperty(window.navigator, 'platform', {
-            value: 'Win32',
-            writable: true,
-        });
-        shallow(<ChannelController {...props}/>);
-
-        expect(document.body.classList.contains('os--windows')).toBe(true);
+    test('Should have app__body and channel-view classes by default', () => {
+        expect(getClassnamesForBody('')).toEqual(['app__body', 'channel-view']);
     });
 
-    test('Should have os--mac class on body for MacIntel', () => {
-        Object.defineProperty(window.navigator, 'platform', {
-            value: 'MacIntel',
-            writable: true,
-        });
-        shallow(<ChannelController {...props}/>);
-
-        expect(document.body.classList.contains('os--mac')).toBe(true);
+    test('Should have os--windows class on body for windows 32 or windows 64', () => {
+        expect(getClassnamesForBody('Win32')).toEqual(['app__body', 'channel-view', 'os--windows']);
+        expect(getClassnamesForBody('Win64')).toEqual(['app__body', 'channel-view', 'os--windows']);
     });
 
-    test('Should have os--mac class on body for MacPPC', () => {
-        Object.defineProperty(window.navigator, 'platform', {
-            value: 'MacPPC',
-            writable: true,
-        });
-        shallow(<ChannelController {...props}/>);
-
-        expect(document.body.classList.contains('os--mac')).toBe(true);
-    });
-
-    test('Should remove app__body and channel-view classes on body on unmount', () => {
-        Object.defineProperty(window.navigator, 'platform', {
-            value: 'MacPPC',
-            writable: true,
-        });
-        const wrapper = shallow(<ChannelController {...props}/>);
-
-        wrapper.unmount();
-        expect(document.body.classList.contains('app__body')).toBe(false);
-        expect(document.body.classList.contains('channel-view')).toBe(false);
+    test('Should have os--mac class on body for MacIntel or MacPPC', () => {
+        expect(getClassnamesForBody('MacIntel')).toEqual(['app__body', 'channel-view', 'os--mac']);
+        expect(getClassnamesForBody('MacPPC')).toEqual(['app__body', 'channel-view', 'os--mac']);
     });
 
     test('Should add .app-bar-enabled class when app bar is enabled', () => {
-        let wrapper = shallow(
+        const {container} = render(
             <ChannelController
-                {...props}
+                fetchingChannels={false}
                 shouldShowAppBar={true}
-            />,
-        );
-        expect(wrapper.find('.channel-view-inner').hasClass('app-bar-enabled')).toBe(true);
+            />);
```

Comment:
Minor nit, but I'd actually expect these closing brackets to be on a new line. I'm not too worried about that though since we haven't got ESLint set up to catch it though

Reply:
i changed nevertheless